### PR TITLE
Nvalluri keysets

### PR DESCRIPTION
**Pull request by valnav**

### CODE REVIEW

**Comment by Lauragra** [comment]
Hi @valnav , please contact me internally regarding this PR. Thanks.

File: api-reference/beta/resources/trustframeworkkeyset.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+PROVIDE DESCRIPTION HERE
+
+## Methods
+
+| Method       | Return Type | Description |
+|:-------------|:------------|:------------|
+| [Get trustFrameworkKeySet](../api/trustframeworkkeyset-get.md) | [trustFrameworkKeySet](trustframeworkkeyset.md) | Read properties and relationships of trustFrameworkKeySet object. |
+| [Update](../api/trustframeworkkeyset-update.md) | [trustFrameworkKeySet](trustframeworkkeyset.md) | Update trustFrameworkKeySet object. |
+| [Delete](../api/trustframeworkkeyset-delete.md) | None | Delete trustFrameworkKeySet object. |
+|[Generatekey](../api/trustframeworkkeyset-generatekey.md)|[trustFrameworkKey](trustframeworkkey.md)||
```
[the remark this answers is not in the document]
Comment (valnav) [reply api-reference/beta/resources/trustframeworkkeyset.md]: added, but please add descriptions to each of the types in trustframeworkkey.md

File: api-reference/beta/api/identityuserflow-update.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+---
+title: "Update identityuserflow"
+description: "Update the properties of identityuserflow object."
+localization_priority: Normal
+author: "valnav"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# Update identityuserflow
```
Comment (agrabhi) [inline api-reference/beta/api/identityuserflow-update.md]: Update operation is not available, please delete this file

File: api-reference/beta/api/identityuserflow-delete.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+---
+title: "Delete UserFlow"
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-delete.md]: The title and H1 below should be "Delete identityUserFlow". Note that we use the resource names and resource names are lower camel case.

File: api-reference/beta/api/identityuserflow-delete.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+author: "valnav"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# Delete UserFlow
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Delete UserFlow.
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-delete.md]: Revise as "Delete an [identityUserFlow] object." And make the resource name a link to the resource topic.

File: api-reference/beta/api/identityuserflow-delete.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+```http
+DELETE /identity/userFlows/{id}
+```
+
+## Request headers
+
+| Name          | Description   |
+|:--------------|:--------------|
+| Authorization | Bearer {token} |
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-delete.md]: Global: This table should indicate that the Authorization header is required. Change the description to "Bearer {token}. Required."

File: api-reference/beta/api/identityuserflow-delete.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+## Request body
+
+Do not supply a request body for this method.
+
+## Response
+
+If successful, this method returns `204, No Content` response code. It does not return anything in the response body.
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-delete.md]: Global for delete topics: revise as "returns a `204 No Content` response code.

File: api-reference/beta/api/identityuserflow-get.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
+---
+title: "Get UserFlow"
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-get.md]: Global: update the resource name to "identityUserFlow" as per previous comment.

File: api-reference/beta/api/identityuserflow-get.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+author: "valnav"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# Get UserFlow
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve the properties and relationships of Userflow object.
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-get.md]: Global: In the summary statements, follow the correct naming/casing for the resource name and add the link to the resource topic. Be sure to add the article (a, an) for readability; "an identityUserFlow object".

File: api-reference/beta/api/identityuserflow-get.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+|:----------|:----------|
+| Authorization | Bearer {token} |
+
+## Request body
+
+Do not supply a request body for this method.
+
+## Response
+
+If successful, this method returns a `200 OK` response code and the requested [UserFlow](../resources/identityuserflow.md) object in the response body.
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-get.md]: Update the resource name here as well.

File: api-reference/beta/api/identityuserflow-post-userflows.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+author: "valnav"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# Create UserFlow
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Use this API to create a new UserFlow.
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-post-userflows.md]: Revise as: "Create a new [identityUserFlow] object." here, and in the description.

File: api-reference/beta/api/identityuserflow-post-userflows.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+```http
+POST /identity/userFlows
+```
+
+## Request headers
+
+| Name          | Description   |
+|:--------------|:--------------|
+| Authorization | Bearer {token} |
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-post-userflows.md]: When a request includes JSON data, we should add the Content-type header to the Request headers table:
|Content-type | application/json. Required. |

File: api-reference/beta/api/identityuserflow-post-userflows.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+
+## Request body
+
+In the request body, supply a JSON representation of [UserFlow](../resources/identityuserflow.md) object.
+
+## Response
+
+If successful, this method returns `201, Created` response code and a new [UserFlow](../resources/identityuserflow.md) object in the response body.
```
Comment (Lauragra) [inline api-reference/beta/api/identityuserflow-post-userflows.md]: `201, Created` -> a `201 Created`

File: api-reference/beta/api/trustframework-list-keysets.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
+---
+title: "List keySets"
+description: "Retrieve a list of trustframeworkkeyset objects."
+localization_priority: Normal
+author: "valnav"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# List keySets
```
Comment (Lauragra) [inline api-reference/beta/api/trustframework-list-keysets.md]: List trustFrameworkKeySets (also update the title)

File: api-reference/beta/api/trustframework-list-keysets.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve a list of trustframeworkkeyset objects.
```
Comment (Lauragra) [inline api-reference/beta/api/trustframework-list-keysets.md]: Update casing of the resource name and add a link.

File: api-reference/beta/api/trustframework-post-keysets.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+author: "valnav"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# Create trustFrameworkKeySet
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Use this API to create a new trustFrameworkKeySet.The id of the trustFrameworkKeySet is expected in the create request. However it can be modified by the service. The modified id will be available in the response and in the location header.
```
Comment (Lauragra) [inline api-reference/beta/api/trustframework-post-keysets.md]: Remove the "Use this API" and start with Create (here, and in the description).

File: api-reference/beta/api/trustframeworkkeyset-generatekey.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+author: "valnav"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# trustFrameworkKeySet: generateKey
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+This api can be used to generate a key and a secret automatically in the keyset. Caller doesn't have to provide a secret here.
```
Comment (Lauragra) [inline api-reference/beta/api/trustframeworkkeyset-generatekey.md]: Revise as: "Generate a key and a secret automatically in the keyset. The caller r doesn't have to provide a secret."

File: api-reference/beta/api/trustframeworkkeyset-generatekey.md [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+| Parameter    | Type        | Description |
+|:-------------|:------------|:------------|
+| `use` | string | Similar to 'use' property of trustFrameworkKey. |
+| `kty` | string | Similar to 'kty' property of trustFrameworkKey. |
+| `nbf` | int | Similar to 'nbf' property of trustFrameworkKey. |
+| `exp` | int | Similar to 'exp' property of trustFrameworkKey. |
+
+## Response
+
+If successful, this method returns `200, OK` response code and a new [trustFrameworkKey](../resources/trustframeworkkey.md) object in the response body.
```
Comment (Lauragra) [inline api-reference/beta/api/trustframeworkkeyset-generatekey.md]: 200, OK -> a 200 OK

File: api-reference/beta/api/trustframeworkkeyset-generatekey.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+|:--------------|:--------------|
+| Authorization | Bearer {token} |
+
+## Request body
+
+In the request body, provide a JSON object with the following parameters.
+
+| Parameter    | Type        | Description |
+|:-------------|:------------|:------------|
+| `use` | string | Similar to 'use' property of trustFrameworkKey. |
```
Comment (Lauragra) [inline api-reference/beta/api/trustframeworkkeyset-generatekey.md]: Remove the tick marks from the parameter names here, for readability in tables.

File: api-reference/beta/api/trustframeworkkeyset-generatekey.md [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+
+```http
+POST /trustFramework/keySets/{id}/generateKey
+```
+
+## Request headers
+
+| Name          | Description   |
+|:--------------|:--------------|
+| Authorization | Bearer {token} |
```
Comment (Lauragra) [inline api-reference/beta/api/trustframeworkkeyset-generatekey.md]: Add the Content-type header.

File: api-reference/beta/api/trustframeworkkeyset-get.md [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+author: "valnav"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# Get trustFrameworkKeySet
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve the properties and relationships of trustframeworkkeyset object.
```
Comment (Lauragra) [inline api-reference/beta/api/trustframeworkkeyset-get.md]: trustframeworkkeyset -> a [trustFrameworkKeySet]

File: api-reference/beta/api/trustframeworkkeyset-update.md [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+|:-----------|:-----------|
+| Authorization | Bearer {token}. Required. |
+| Content-type  | application/json. Required.|
+
+## Request body
+
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|keys|trustFrameworkKey collection||
```
Comment (Lauragra) [inline api-reference/beta/api/trustframeworkkeyset-update.md]: Add the missing description here.

File: api-reference/beta/resources/identityuserflow.md [hunk 22]
```diff
@@ -0,0 +1,67 @@
+---
+title: "UserFlow resource type"
+description: "PROVIDE DESCRIPTION HERE"
```
Comment (Lauragra) [inline api-reference/beta/resources/identityuserflow.md]: Add the missing description here and in the summary. Also, the resource name should be identityUserFlow, right?

File: api-reference/beta/resources/identityuserflow.md [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+PROVIDE DESCRIPTION HERE
+
+## Methods
+
+| Method       | Return Type | Description |
+|:-------------|:------------|:------------|
+| [List](../api/identityuserflow-list.md) | [UserFlow](identityuserflow.md) | List UserFlows. |
```
Comment (Lauragra) [inline api-reference/beta/resources/identityuserflow.md]: Make the descriptions here match the summary sentence in the API topics.

File: api-reference/beta/resources/trustframeworkkey.md [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents a JWK (JSON Web Key). TrustFrameworkKey is a JSON data structure that represents a cryptographic key. The structure of this resource follows the format defined in [RFC 7517 Section 4](https://tools.ietf.org/html/rfc7517#section-4)
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+| `kid` | string | The unique identifier for the key.   |
```
Comment (Lauragra) [inline api-reference/beta/resources/trustframeworkkey.md]: Remove the tick marks from the property names. In text, property names are bold, but in tables, we don't apply formatting for better readability.

File: api-reference/beta/resources/trustframeworkkeyset.md [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents a trust framework key set/policy keys. The Identity Experience framework stores the secrets, which can be used in the policies. The secrets can be passwords, certificates, or other files. In the portal, these entities are shown as `Policy keys`. The Identity Experience framework uses the JSON Web Key (JWK) standard for the key sets. This entity follows the format specified in [RFC 7517 Section 5](https://tools.ietf.org/html/rfc7517#section-5).
+
+## Methods
+
+| Method       | Return Type | Description |
+|:-------------|:------------|:------------|
+| [Get trustFrameworkKeySet](../api/trustframeworkkeyset-get.md) | [trustFrameworkKeySet](trustframeworkkeyset.md) | Read properties and relationships of trustFrameworkKeySet object. |
```
Comment (Lauragra) [inline api-reference/beta/resources/trustframeworkkeyset.md]: Make sure that the method descriptions match the descriptions in the API topics. Make the link text consistent, i.e. include the resource name.

File: api-reference/beta/toc.yml [hunk 26]
```diff
@@ -8826,7 +8826,39 @@
           href: api/trustframework-put-trustframeworkpolicy.md
         - name: Delete
           href: api/trustframeworkpolicy-delete.md
-    - name: Mail
+      - name: Trust Framework Keyset
```
Comment (Lauragra) [inline api-reference/beta/toc.yml]: Is keyset one work or two? In the content it's "key set". Also, is this term generally title capped? If not, use sentence casing: Trust framework key set.

File: concepts/changelog.md [hunk 27]
```diff
@@ -17,6 +17,9 @@ For details about known issues with Microsoft Graph APIs, see [Known issues](kno
 
 | **Change type** | **Version** | **Description**                  |
 |:----------------|:------------|:-----------------------------------------|
+
+| Addition | Beta | Added the **Trustframeworkkeyset** property to the [Trustframeworkkeyset](/graph/api/resources/trustframeworkkeyset?view=graph-rest-beta) resource. |
```
Comment (Lauragra) [inline concepts/changelog.md]: The changelog entry should mention the new resource and operations, yes? You don't have to list/link to each operation, you can say "and associated operations", for example.

File: api-reference/beta/toc.yml (hunk 26 above)
Comment (valnav) [reply api-reference/beta/toc.yml]: keyset is one word

**Comment by valnav** [comment]
> Please review and incorporate editorial feedback.

can you please re-review

File: api-reference/beta/api/identityuserflow-get.md (hunk 7 above)
Comment (valnav) [reply api-reference/beta/api/identityuserflow-get.md]: checked with devs, we want to go with UserFlow

**Comment by valnav** [comment]
> Please review and incorporate editorial feedback.

finished with the updates

**Comment by valnav** [comment]
> Please
> 
> * separate out user flow PR.
> * remove update file from user flow.

agreed to keep it in the same PR

**Pull request merged**